### PR TITLE
chore(example): add finetune example for stable diffusion

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

Add the example as following steps:
1. build origin model package
2. build eval and fine-tune dataset
3. build fine-tune model package and generate new ft model package
4. compare the origin and ft results：http://e2e-20231023.pre.intra.starwhale.ai/simple/report/preview/?rid=3f7ac487-7225-4eca-a687-1aea33e9d982

**TODO**

1. Support dataset head when run a handler in web ui.
2. Supports blood relationships of artifacts.
3. How to identify a fine-tune job?
4. Support model file edit?
5. Export a report?

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [x] Others

## Checklist
- [x] run code format and lint check
- [ ] add unit test
- [x] add necessary doc

### CODE REVIEW

**Comment by tianweidut** [comment]
Combine `[build_dataset_for_eval.py](https://github.com/star-whale/starwhale/pull/2641/files#diff-d68daf33005c87c31ee200ab7afa7e1c9d33582afd0645032e820f30062d3c99)` and `example/stable-diffusion/txt2img-ft/build_dataset_for_ft.py` into one script? @goldenxinxing

**Comment by tianweidut** [comment]
upload example/stable-diffusion/txt2img-ft/report.png into private aliyun oss storage, and copy the public download link?

File: example/stable-diffusion/txt2img-ft/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+accelerate>=0.16.0
+torchvision==0.15.2
+transformers>=4.25.1
+datasets==2.13.0
+diffusers==0.17.1
+ftfy
+tensorboard
+Jinja2
+gradio
```
Comment (tianweidut) [inline example/stable-diffusion/txt2img-ft/requirements.txt:9]: remove gradio dep?  starwhale[online-server] has already installed gradio.

File: example/stable-diffusion/txt2img-ft/runtime.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+api_version: 1.1
+dependencies:
+  - pip:
```
Comment (tianweidut) [inline example/stable-diffusion/txt2img-ft/runtime.yaml:3]: remove `starwhale>=0.6.0`? 
and in the req.txt, `starwhale[online-serve] >= 0.6.0`

**Comment by tianweidut** [comment]
add these example into https://cloud.starwhale.cn/projects/374/overview ? @goldenxinxing

File: example/stable-diffusion/txt2img-ft/build_dataset_for_eval.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                "text": line,
+            }
+        )
+    ds.commit()
+    ds.close()
+    print("build done!")
+
+
+def build_ds_from_hf(ds_uri, dataset_name: str = "lambdalabs/pokemon-blip-captions"):
```
[the remark this answers is not in the document]
Comment (goldenxinxing) [reply example/stable-diffusion/txt2img-ft/build_dataset_for_eval.py]: done

**Comment by goldenxinxing** [comment]
> upload example/stable-diffusion/txt2img-ft/report.png into private aliyun oss storage, and copy the public download link?

done

File: example/stable-diffusion/txt2img-ft/requirements.txt (hunk 1 above)
Comment (goldenxinxing) [reply example/stable-diffusion/txt2img-ft/requirements.txt]: done

File: example/stable-diffusion/txt2img-ft/runtime.yaml (hunk 2 above)
Comment (goldenxinxing) [reply example/stable-diffusion/txt2img-ft/runtime.yaml]: done

**Comment by goldenxinxing** [comment]
> add these example into https://cloud.starwhale.cn/projects/374/overview ? @goldenxinxing

done, https://cloud.starwhale.cn/simple/report/preview/?rid=33b8c99d-c9bf-4631-9970-6c7a8bcff6bd

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**